client/src/pages/CreateSurvey/Questions/Questions.jsx

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable react/jsx-wrap-multilines */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  TextField,
+  Button,
+  Select,
+  InputLabel,
+  FormControl,
+  MenuItem,
+  Checkbox,
+  Box,
+  FormControlLabel,
+  Typography,
+  Divider,
+} from '@material-ui/core';
+import DeleteIcon from '@material-ui/icons/Delete';
+import Options from '../Options/Options';
+
+// TODO FIX WEIRD LINTING ERRORS
+// TODO fix annoying state update typing thing
+const Question = ({ questionIndex }) => {
+  const { title, type, required, commentsEnabled } = useSelector(
+    (state) => state.createSurveyReducer.questions[questionIndex],
+  );
+
+  const dispatch = useDispatch();
+
+  const setQuestionData = (event, inputType) => {
+    event.persist();
+    const payload = {};
+    payload.index = questionIndex;
+    payload[event.target.name] =
+      inputType === 'checkbox' ? event.target.checked : event.target.value;
+    dispatch({ type: 'SET_QUESTION_DATA', payload });
+  };
+
+  const handleDeleteQuestion = () => {
+    const payload = {};
+    payload.index = questionIndex;
+    dispatch({ type: 'DELETE_QUESTION', payload });
+  };
+
+  return (
```

Comment:
Return statement has a lot going on in it and quite hard to digest, I think it would require breaking down into const's which describe what each section is for.